edweiss-app/app/(app)/deck/[id]/card/creation/index.tsx

```diff
@@ -37,95 +38,111 @@ import { DecksRepository } from '../../../_layout';
  * @returns {ApplicationRoute} Screen to create a card
  */
 const CreateCardScreen: ApplicationRoute = () => {
-	const { deckId } = useStringParameters();
-	const [question, setQuestion] = useState("");
-	const [answer, setAnswer] = useState("");
-	const [existedQuestion, setExistedQuestion] = useState(false);
-
-	// const deck = useDoc(Collections.deck, deckId);
-
-	const [deck, handler] = useRepositoryDocument(deckId, DecksRepository);
-
-	// if (!deck)
-	// 	return <Redirect href={'/'} />;
-
-	// Create a new card
-	async function createCard() {
-
-		if (question.length == 0 || answer.length == 0 || existedQuestion)
-			return;
-
-		const isDuplicate = deck?.data.cards.some(card => card.question === question);
-		if (isDuplicate) {
-			setExistedQuestion(true);
-			return;  // Prevent creation if a duplicate is found
-		}
-
-		const card: Memento.Card = {
-			question: question,
-			answer: answer,
-			learning_status: "Not yet",
-		};
-
-		const previousCards = deck?.data.cards ?? [];
-
-		handler.modifyDocument(deckId, { cards: [...previousCards, card] }, () => {
-			callFunction(Memento.Functions.createCard, {
-				deckId: deckId,
-				card
-			});
-		});
-
-		// if (res.status == 1) {
-		// 	console.log(`OKAY, card created with id ${res.data.id}`);
-		// }
-
-		router.back();
-	}
-
-	return (
-		<>
-			<RouteHeader title='Create a card' />
-
-			<TScrollView>
-				{/* <TextInput value={deckName} onChangeText={n => setDeckName(n)} placeholder='Deck name' placeholderTextColor={'#555'} style={{ backgroundColor: Colors.dark.crust, borderColor: Colors.dark.blue, borderWidth: 1, padding: 8, paddingHorizontal: 16, margin: 16, marginBottom: 0, color: 'white', borderRadius: 14, fontFamily: "Inter" }}>
+    const { deckId } = useStringParameters();
```

Comment:
Out of curiosity, I am seeing a huge chunk of change (red color), but the new code seems mostly similar (except for checks). Do you know why is that?

Reply:
There is a commented code in { /* ... */} so I just deleted it I think.

Reply:
Ah so first I added some "useStates", then I adapted optimistic update feature. There are many changes so the editor probably assume that I just deleted everything before and make something totally new.

Reply:
Ah I see, one of the thing is the indentation is not the same.